pcsx2/Recording/InputRecording.cpp

```diff
@@ -91,58 +91,57 @@ void InputRecording::ControllerInterrupt(u8& data, u8& port, u16& bufCount, u8 b
 
 	if (bufCount == 1)
 		fInterruptFrame = data == READ_DATA_AND_VIBRATE_FIRST_BYTE;
-	else if (bufCount == 2 && buf[bufCount] != READ_DATA_AND_VIBRATE_SECOND_BYTE)
-		fInterruptFrame = false;
-
-	// We do not want to record or save the first two bytes in the data returned from the PAD plugin
-	else if (fInterruptFrame && bufCount >= 3 && frameCounter >= 0 && frameCounter < INT_MAX)
+	else if (bufCount == 2)
+	{
+		if (buf[bufCount] != READ_DATA_AND_VIBRATE_SECOND_BYTE)
+			fInterruptFrame = false;
```

Comment:
I don't see the need for a nested if, can just be an AND like before

Reply:
Because regardless of if `buf[bufCount] is equal to READ_DATA_AND_VIBRATE_SECOND_BYTE`, bufCount is *still* 2. Therefore, the very next else if (originally) would already be false before checking its conditions - `fInterruptFrame && bufCount >= 3 && frameCounter >= 0 && frameCounter < INT_MAX`.
By nesting the `buf[bufCount] != READ_DATA_AND_VIBRATE_SECOND_BYTE`, we negate the need for `bufCount >= 3`.